library/core/src/main/java/com/google/android/exoplayer2/drm/HttpMediaDrmCallback.java

```diff
@@ -47,29 +47,44 @@
   }
 
   private final HttpDataSource.Factory dataSourceFactory;
-  private final String defaultUrl;
+  private final String defaultLicenseUrl;
+  private final boolean forceDefaultLicenseUrl;
   private final Map<String, String> keyRequestProperties;
 
   /**
-   * @param defaultUrl The default license URL.
+   * @param defaultLicenseUrl The default license URL.
    * @param dataSourceFactory A factory from which to obtain {@link HttpDataSource} instances.
    */
-  public HttpMediaDrmCallback(String defaultUrl, HttpDataSource.Factory dataSourceFactory) {
-    this(defaultUrl, dataSourceFactory, null);
+  public HttpMediaDrmCallback(String defaultLicenseUrl, HttpDataSource.Factory dataSourceFactory) {
+    this(defaultLicenseUrl, false, dataSourceFactory, null);
   }
 
   /**
-   * @deprecated Use {@link HttpMediaDrmCallback#HttpMediaDrmCallback(String, Factory)}. Request
-   *     properties can be set by calling {@link #setKeyRequestProperty(String, String)}.
-   * @param defaultUrl The default license URL.
+   * @param defaultLicenseUrl The default license URL.
+   * @param forceDefaultLicenseUrl Whether to force use of {@code defaultLicenseUrl} even for key
+   *     requests that include their own license URL.
+   * @param dataSourceFactory A factory from which to obtain {@link HttpDataSource} instances.
+   */
+  public HttpMediaDrmCallback(String defaultLicenseUrl, boolean forceDefaultLicenseUrl,
+      HttpDataSource.Factory dataSourceFactory) {
+    this(defaultLicenseUrl, forceDefaultLicenseUrl, dataSourceFactory, null);
+  }
+
+  /**
+   * @deprecated Use {@link HttpMediaDrmCallback#HttpMediaDrmCallback(String, boolean, Factory)}.
```

Comment:
There's not much point keeping this method, given it's deprecated and you're changing the signature (a caller will need to change their code one way or another, so they should just start using the non-deprecated ones ;)). Can we remove it and merge the body into the one above?